ConsoleApp1/Program.cs

```diff
@@ -1,9 +1,361 @@
+                        break;
+                    case ConsoleKey.D0:
+                    default:
+                        Environment.Exit(0);
+                        break;
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Program error occured: {ex.GetType()} {Environment.NewLine} {ex.Message} ");
+            }
+            Console.WriteLine("Press any key to continue");
+            Console.ReadKey();
+            Console.Clear();
+        }
+
+        static (string, string, string)[] ReadPhoneBookRecords(string path)
+        {
+            Console.WriteLine();
+            string[] records;
+            try
+            {
+                records = File.ReadAllLines(path);
+            }
+            catch
+            {
+                throw;
+            }
+            
+            //FileNotFoundException 
+            (string, string, string)[] phoneBookRecords = new (string, string, string)[records.Length - 1];
+
+            for (int i = 1; i < records.Length; i++)
+            {
+                var itemProperties = records[i].Split(',');
+
+                phoneBookRecords[i - 1] = (itemProperties[0], itemProperties[1], itemProperties[2]);
+            }
+
+            return phoneBookRecords;
+        }
+
+        static void DisplayPhoneBook((string firstName, string lastName, string phone)[] records)
+        {
+            Console.WriteLine();
+            Console.WriteLine($"{"First Name",-15} {"Last Name",-15} {"Phone",-15}");
+
+            foreach (var record in records)
+            {
+                Console.WriteLine($"{record.firstName,-15} {record.lastName,-15} {record.phone,-15}");
+            }
+        }
+
+        static void AddNewRecord(string path)
+        {
+            
+            Console.Write("First Name: ");
+            var firstName = Console.ReadLine();
+
+            Console.Write("Last Name: ");
+            var lastName = Console.ReadLine();
+
+            Console.Write("Phone: ");
+            var phone = Console.ReadLine();
+            string[] newRecords;
+            try
+            {
+                newRecords = new[] { string.Join(',', new[] { firstName, lastName, phone }) };
+
+                File.AppendAllLines(path, newRecords);
+            }
+            catch (NullReferenceException ex)
+            {
+                Console.WriteLine($"{ex.Message}");
+                throw;
+            }
+        }
+
+        static void SearchRecords(string path)
+        {
+            Console.WriteLine();
+            Console.WriteLine("Enter search string: ");
+            string searchText = Console.ReadLine() ?? "";
+            (string firstName, string lastName, string phone)[] phoneBookRecords;
+            
+            try
+            {
+                phoneBookRecords = ReadPhoneBookRecords(path);
+            }
+            catch (NullReferenceException ex)
```

Comment:
NullReferenceException can not be here only inside ReadPhoneBookRecords, so move that error handler inside the method